src/server/src/main/java/io/cassandrareaper/service/Heart.java

```diff
@@ -185,6 +213,44 @@ private void grabAndStoreNodeMetrics(IDistributedStorage storage, UUID runId, No
             .build());
   }
 
+  private void grabAndStoreGenericMetrics()
+      throws ReaperException, InterruptedException, JMException {
+    Node node = Node.builder().withClusterName(context.localClusterName).withHostname(context.localNodeAddress).build();
+    List<GenericMetric> metrics
+        = this.metricsService.convertToGenericMetrics(
+            ClusterFacade.create(context).collectMetrics(node, COLLECTED_METRICS), node);
+
+    for (GenericMetric metric:metrics) {
+      ((IDistributedStorage)context.storage).storeMetric(metric);
+    }
+    LOG.debug("Grabbing and storing metrics for {}", context.localNodeAddress);
+
+  }
+
+  private void grabAndStoreActiveCompactions()
+      throws JsonProcessingException, MalformedObjectNameException, ReflectionException,
+          ReaperException, InterruptedException {
+    Node node = Node.builder().withClusterName(context.localClusterName).withHostname(context.localNodeAddress).build();
+    List<Compaction> activeCompactions = ClusterFacade.create(context).listActiveCompactionsDirect(node);
+
+    ((IDistributedStorage) context.storage)
+        .storeCompactions(context.localClusterName, context.localNodeAddress, activeCompactions);
+
+    LOG.debug("Grabbing and storing compactions for {}", context.localNodeAddress);
+  }
+
+  private void grabAndStoreActiveStreams()
+      throws JsonProcessingException, MalformedObjectNameException, ReflectionException,
+          ReaperException, InterruptedException {
+    Node node = Node.builder().withClusterName(context.localClusterName).withHostname(context.localNodeAddress).build();
+    Set<CompositeData> activeStreams = ClusterFacade.create(context).listStreamsDirect(node);
+
+    ((IDistributedStorage) context.storage)
+        .storeStreams(context.localClusterName, context.localNodeAddress, activeStreams);
+
+    LOG.debug("Grabbing and storing streams for {}", context.localNodeAddress);
+  }
```

Comment:
can we move these 3 methods down in to `MetricsService` ?

Reply:
Sure thing, makes sense